libp2p/stream_muxer/mplex/mplex.py

```diff
@@ -1,25 +1,49 @@
 import asyncio
+from typing import TYPE_CHECKING, Tuple, Dict
 
-from .utils import encode_uvarint, decode_uvarint_from_stream, get_flag
+from libp2p.stream_muxer.muxed_connection_interface import IMuxedConn
+
+from .constants import HeaderTags
+from .utils import encode_uvarint, decode_uvarint_from_stream
 from .mplex_stream import MplexStream
-from ..muxed_connection_interface import IMuxedConn
+
+
+if TYPE_CHECKING:
+    from multiaddr import Multiaddr
+    from libp2p.security.secure_conn_interface import ISecureConn
```

Comment:
I've found that this circular import problem is fixed with a `libp2p.abc` module like [this from py-evm](https://github.com/ethereum/py-evm/blob/3b0bcc21954134e0293e40b1850e6bf16e5f4c27/eth/abc.py) allow you to have good interface enforced type hints that don't cause circular imports.